### ISSUE TITLE
Stalking and Harrasment

### ISSUE THREAD

**Issue by Machinelearning1954** [Machinelearning1954/Facial-recognition-#5]
#!/usr/bin/env python3

"""Loads the Labeled Faces in the Wild (LFW) dataset using scikit-learn."""

from sklearn.datasets import fetch_lfw_people
import os
import joblib

# Define where to cache the dataset
# Using the default scikit-learn data home is usually fine,
# but specifying it explicitly for clarity within the project.
# Note: fetch_lfw_people uses joblib which caches by default in ~/scikit_learn_data
# We'll let it use the default cache location for simplicity.

def load_lfw_dataset(min_faces=70, resize=0.4):
    """Loads the LFW dataset, downloading if necessary.

    Args:
        min_faces (int): Minimum number of faces per person to include.
        resize (float): Ratio to resize the images.

    Returns:
        Bunch: The LFW dataset object from scikit-learn.
    """
    print(f"Loading LFW dataset (min_faces_per_person={min_faces}, resize={resize})...")
    print("This may take several minutes the first time as the data is downloaded (~200MB) and processed.")
    
    # Fetch the dataset. It will be downloaded and cached automatically
    # in the user's home directory (e.g., ~/scikit_learn_data/lfw_home/)
    lfw_people = fetch_lfw_people(min_faces_per_person=min_faces, resize=resize)
    
    n_samples, h, w = lfw_people.images.shape
    n_features = lfw_people.data.shape[1]
    n_classes = lfw_people.target_names.shape[0]

    print("\nDataset loaded successfully.")
    print(f"Total dataset size:")
    print(f"  Number of samples: {n_samples}")
    print(f"  Number of features (pixels): {n_features}")
    print(f"  Number of classes (people): {n_classes}")
    print(f"  Image dimensions: {h}x{w}")
    
    return lfw_people

if __name__ == "__main__":
    # Example usage: Load the dataset with default parameters
    dataset = load_lfw_dataset()
    print("\nFirst 5 target names (people):")
    print(dataset.target_names[:5])
    print("\nFirst 5 target labels (indices):")
    print(dataset.target[:5])
